### DISCUSSION TITLE
Matplotlib (Tkinter) doesn't work on Windows instances

### DISCUSSION THREAD

**Discussion by ma-sadeghi** [community/community#26434] [Actions]
<p>I’m using GitHub Actions for CI in one of my projects (Python). In of of the unit tests, something is plotted using Matplotlib library. The workflow runs smoothly on Ubuntu and macOS, but fails on Windows, and it seems that the error is coming from that unit test and is related to Tkinter. Here’s the error I’m getting:</p>
<pre><code>================================== FAILURES ===================================
_________________[doctest] openpnm.network.Delaunay.Delaunay_________________
059 Supplying custom specified points:
060
061 &gt;&gt;&gt; pts = sp.rand(200, 3)
062 &gt;&gt;&gt; gn = op.network.Delaunay(points=pts, shape=[1, 1, 1])
063 &gt;&gt;&gt; gn.Np
064 200
065
066 Which can be quickly visualized using:
067
068 &gt;&gt;&gt; fig = op.topotools.plot_connections(network=gn)
UNEXPECTED EXCEPTION: TclError('Can\'t find a usable init.tcl in the following directories: \n {c:\\hostedtoolcache\\windows\\python\\3.6.8\\x64\\tcl\\tcl8.6}\n\nc:/hostedtoolcache/windows/python/3.6.8/x64/tcl/tcl8.6/init.tcl: couldn\'t read file "c:/hostedtoolcache/windows/python/3.6.8/x64/tcl/tcl8.6/init.tcl": No error\ncouldn\'t read file "c:/hostedtoolcache/windows/python/3.6.8/x64/tcl/tcl8.6/init.tcl": No error\n while executing\n"source c:/hostedtoolcache/windows/python/3.6.8/x64/tcl/tcl8.6/init.tcl"\n ("uplevel" body line 1)\n invoked from within\n"uplevel #0 [list source $tclfile]"\n\n\nThis probably means that Tcl wasn\'t installed properly.\n',)
Traceback (most recent call last):

  File "c:\hostedtoolcache\windows\python\3.6.8\x64\lib\doctest.py", line 1330, in __run
    compileflags, 1), test.globs)

  File "&lt;doctest openpnm.network.Delaunay.Delaunay[5]&gt;", line 1, in &lt;module&gt;
</code></pre>
<p>I’d really appreciate it if someone could shed some light. </p>
<p>Many thanks,</p>
<p>Amin</p>

**Comment by moux233** [comment]
<p>Hi Amin</p>
<p>I’ve had the exact same error, using pytest.</p>
<p>Did you find a solution?</p>
<p>Maxime</p>

**Comment by ma-sadeghi** [comment, marked as answer]
<p>Hi Maxime,</p>
<p>Yes I did, you need to explicitly define the <code>matplotlib</code> backend prior to plotting. Here’s the <a href="https://github.com/PMEAL/OpenPNM/commit/8060cdc0f49c6735301a7fd834c43044eee56ab9" rel="nofollow noopener">commit</a> that seemed to fix the problem in our case.</p>
<p>Best,</p>
<p>Amin</p>
